app/agents/ApplicationAnalyzerAgent.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import random
+import spade.behaviour
+
+from app.agents.RecruitmentManagerAgent import RecruitmentManagerAgent
+from app.dataaccess.model.JobOffer import ApplicationStatus
+from app.dataaccess.model.MessageType import MessageType
+from app.modules.JobOfferModule import JobOfferModule
+from app.utils.configuration import MASConfiguration
+
+from .base.BaseAgent import BaseAgent
+
+
+class ApplicationAnalyzerAgent(BaseAgent):
+    _instances: int = 0
+
+    def __init__(
+        self,
+    ):
+        ApplicationAnalyzerAgent._instances = ApplicationAnalyzerAgent._instances + 1
+        super().__init__(f"{ApplicationAnalyzerAgent._instances}")
+
+        self.rmentJID = self.config.agents[RecruitmentManagerAgent.__name__.split('.')[-1]].jid
+
+        self.jobOfferModule = JobOfferModule(self.agent_config.dbname, self.logger)
+
+        # behaviours
+        self.analyzeBehav: Analyze = None
+
+    async def setup(self):
+        await super().setup()
+
+        self.analyzeBehav = Analyze()
+        self.add_behaviour(self.analyzeBehav)
+
+
+class Analyze(spade.behaviour.CyclicBehaviour):
+    """
+    Awaits application analyze request, performs analysis and sends rate.
+    Protocols/Activities in GAIA (role ApplicationAnalyzer): AnalyzeRequest, RateCandidate, AnalyzeResponse
+    """
+
+    agent: ApplicationAnalyzerAgent
+
+    async def run(self):
+        self.agent.logger.info("Analyze behaviour run.")
+
+        # AnalyzeRequest protocol
+        msg = await self.receive(timeout=30)
+        if msg is None:
+            return
+        
+        self.agent.logger.info("Received analyze request: %s", msg.body)
```

Comment:
Raczej, z tego co wiem, to preferowany sposób pisania tego w Pythonie to używanie f-stringów, czyli: `f"Received analyze request: {msg.body}"`. Podobno jest to szybsze XD

Reply:
Dziwne xD

Reply:
poprawiłem na swoim branchu zamykam.